Feature-ListAppointments

Niravanaa opened the pull request:
### Proposed Changes:
This commit contains the initial added files to integrate the Retrofit dependency into the project. The dependency performs an HTTP GET request to a test Firebase database that I have setup to obtain some sample/dummy appointment data.

### Details:
- Added Dependency to build.gradle file
- Added Internet permission to Manifest file
- Added an AppointmentAdapter
- Setup the logic to fetch and display the appointments in the "List Appointments" View
- Modified `MainActivity.kt` to have three properly functioning buttons that switch between the different fragments (views)
- Added an `Appointment.kt` model that has some basic properties (to be verified later)
- Added an `item_border.xml` file to display borders around appointment items
- Adjusted the string placeholders in `strings.xml`.
- Added version number in `gradle.properties`.

### Related Issue(s):
- Closes #6

Niravanaa committed:
Commit1-ListAppointmentsFeature

### Commit Description:
This commit contains the initial added files to integrate the Retrofit dependency into the project. The dependency performs an HTTP GET request to a test Firebase database that I have setup to obtain some sample/dummy appointment data.

### Details:
- Added Dependency to build.gradle file
- Added Internet permission to Manifest file
- Added an AppointmentAdapter
- Setup the logic to fetch and display the appointments in the "List Appointments" View
- Modified `MainActivity.kt` to have three properly functioning buttons that switch between the different fragments (views)
- Added an `Appointment.kt` model that has some basic properties (to be verified later)
- Added an `item_border.xml` file to display borders around appointment items
- Adjusted the string placeholders in `strings.xml`.
- Added version number in `gradle.properties`.

### Related Issue(s):
- Closes #10

Niravanaa committed:
Merge branch 'development' of https://github.com/priyank1968/CloverEmployeeAppointment into Nirav-Issue6Feature

priyank1968 commented on app/src/main/res/layout/item_appointment.xml:
```diff
@@ -0,0 +1,36 @@
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textStyle="bold"
+        android:textSize="18sp"
```
Add these dimensions in dimen.xml and reference them. Eg: ln#13 https://github.com/priyank1968/CloverEmployeeAppointment/blob/a8b1189b5028f04038081c631fe3fbe2fc6e3ddf/app/src/main/res/layout/activity_main.xml#L13

Niravanaa replied:
As seen in [7a91bc1](https://github.com/priyank1968/CloverEmployeeAppointment/pull/19/commits/7a91bc105cdc8e5076febd0d63dec3df4ab56dd6), refactored the hard-coded values into `dimen.xml`.

priyank1968 commented on app/src/main/java/com/example/employeeappointmentmaker/AppointmentAdapter.kt:
```diff
@@ -0,0 +1,40 @@
+    RecyclerView.Adapter<AppointmentAdapter.AppointmentViewHolder>() {
+
+    inner class AppointmentViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        val titleTextView: TextView = itemView.findViewById(R.id.textViewTitle)
```
We should stick to using viewbinding rather than using R.id to reference. It is better to stick with view binding styling rather than linking layout with logic as it is little outdated

Niravanaa replied:
As seen in [7a91bc1](https://github.com/priyank1968/CloverEmployeeAppointment/pull/19/commits/7a91bc105cdc8e5076febd0d63dec3df4ab56dd6), switched the item getting to viewbinding rather than R.id referencing.

priyank1968 commented on app/src/main/res/drawable/item_border.xml:
```diff
@@ -0,0 +1,12 @@
+    <solid android:color="#FFFFFF" />
+    <stroke
+        android:width="1dp"
+        android:color="#CCCCCC" />
```
similar to dimens we want to keep the color hex values also stored in colors.xml and refrence them.

Niravanaa replied:
As seen in [7a91bc1](https://github.com/priyank1968/CloverEmployeeAppointment/pull/19/commits/7a91bc105cdc8e5076febd0d63dec3df4ab56dd6), refactored the hard-color hex-codes into `color.xml`.

priyank1968 commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -31,17 +29,27 @@
     </LinearLayout>
 
+    <Button
+        android:id="@+id/buttonList"
```
This is one for the future but we will create style components for the buttons so it is just plug and play. No changes needed here right now

Niravanaa replied:
As seen in [7a91bc1](https://github.com/priyank1968/CloverEmployeeAppointment/pull/19/commits/7a91bc105cdc8e5076febd0d63dec3df4ab56dd6), removed the button to list appointments. Though I am not sure how to go back to the list appointments fragment.

priyank1968 reviewed (commented):
Take a look at these comments

Niravanaa committed:
fix: First Set of Comments

### Commit Description:
Modified code after first set of comments provided by Priyank. Also added a new Github Action and removed the old Github Action.

### Details:
- Added a new Github Action
- Removed the old Github Action
- Modified the code to use databinding and not R.id finding
- Refactored the hard-coded sizing into `dimen.xml`
- Refactoring the hard-coded colors into `colors.xml`

### Related Issue(s):
- Closes #6

priyank1968 commented on .github/workflows/automaticSemVer.yml:
```diff
@@ -0,0 +1,26 @@
+      - name: Semantic Release
+        run: npx semantic-release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Should this file be part of the commit?

Niravanaa replied:
Yes, this GitHub Action is a better alternative to the previous versioning action that I have told you about previously.

This one will analyze the commits of PRs and determine the next best version to release. I forgot to add a configuration file for it as well, which I will do in this next commit.

priyank1968 commented on app/src/main/res/drawable/item_border.xml:
```diff
@@ -0,0 +1,11 @@
+        android:width="1dp"
+        android:color="@color/lightGrey" />
+    <padding
+        android:left="4dp"
```
Missed the dimens for these

Niravanaa replied:
These dimensions will be refactored into `dimen.xml`.

priyank1968 commented on app/src/main/res/layout/fragment_appointment_list.xml:
```diff
@@ -1,15 +1,24 @@
-    android:layout_height="match_parent"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:padding="16dp"
```
Missed the dimens. You can create double padding value in dimens or can use double margin. Will leave it up to you

Niravanaa replied:
Added the 16dp dimen as well.

priyank1968 reviewed (approved):
Great Job! Just take a look at the comments I left and merge if you with/without suggested changes.

Niravanaa committed:
fix: Refactored Dimension Values

### Commit Description:
This commit refactors certain dimension values.

### Details:
- Refactored dimension values in the fragment view `.xml` file.
- Refactored the dimension values in the `item_border.xml` file.
- Added a configuration file for the Semantic Release integration.

### Related Issue(s):
- Closes #6
